geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/iterators/KryoLazyFilterTransformIterator.scala

```diff
@@ -0,0 +1,120 @@
+/***********************************************************************
+* Copyright (c) 2013-2015 Commonwealth Computer Research, Inc.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Apache License, Version 2.0 which
+* accompanies this distribution and is available at
+* http://www.opensource.org/licenses/apache2.0.php.
+*************************************************************************/
+
+package org.locationtech.geomesa.accumulo.iterators
+
+import java.util.{Collection => jCollection, Map => jMap}
+
+import com.typesafe.scalalogging.slf4j.Logging
+import org.apache.accumulo.core.client.IteratorSetting
+import org.apache.accumulo.core.data.{ByteSequence, Key, Range, Value}
+import org.apache.accumulo.core.iterators.{IteratorEnvironment, SortedKeyValueIterator}
+import org.geotools.filter.text.ecql.ECQL
+import org.locationtech.geomesa.features.kryo.{KryoBufferSimpleFeature, KryoFeatureSerializer}
+import org.locationtech.geomesa.filter.factory.FastFilterFactory
+import org.locationtech.geomesa.utils.geotools.SimpleFeatureTypes
+import org.opengis.feature.simple.SimpleFeatureType
+import org.opengis.filter.Filter
+
+/**
+ * Iterator that operates on kryo encoded values. It will:
+ *   a) evaluate the feature against an optional filter
+ *   b) apply an optional transform
+ *
+ * Uses lazy evaluation of attributes and binary transforms when possible.
+ */
+class KryoLazyFilterTransformIterator extends SortedKeyValueIterator[Key, Value] with Logging {
```

Comment:
this class is moved from the geomesa-accumulo-filter-function module, but not changed.